docs/src/pages/components/icons/icons.md

```diff
@@ -40,28 +40,62 @@ The configuration of a Babel plugin is encouraged.
 
-- `Filled` "theme" (default) is exported as `@material-ui/icons/Delete`,
-- `Outlined` "theme" is exported as `@material-ui/icons/DeleteOutlined`,
-- `Rounded` "theme" is exported as `@material-ui/icons/DeleteRounded`,
-- `Two tone` "theme" is exported as `@material-ui/icons/DeleteTwoTone`,
-- `Sharp` "theme" is exported as `@material-ui/icons/DeleteSharp`.
+- `Filled` "theme" (default) is exported as *@material-ui/icons/Delete*,
+- `Outlined` "theme" is exported as *@material-ui/icons/DeleteOutlined*,
+- `Rounded` "theme" is exported as *@material-ui/icons/DeleteRounded*,
+- `Two tone` "theme" is exported as *@material-ui/icons/DeleteTwoTone*,
+- `Sharp` "theme" is exported as *@material-ui/icons/DeleteSharp*.
 
 Note: The Material Design specification names the icons using "snake_case" naming (for example `delete_forever`, `add_a_photo`), while `@material-ui/icons` exports the respective icons using "PascalCase" naming (for example `DeleteForever`, `AddAPhoto`). There are three exceptions to this naming rule: `3d_rotation` exported as `ThreeDRotation`, `4k` exported as `FourK`, and `360` exported as `ThreeSixty`.
 
 {{"demo": "pages/components/icons/SvgMaterialIcons.js"}}
 
 ## SvgIcon
 
-If you need a custom SVG icon (not available in Material Icons) you should use the `SvgIcon` wrapper.
-The `SvgIcon` component takes the SVG `path` element as its child and converts it to a React component that displays this SVG icon, and allows the icon to be styled and respond to mouse events.
-SVG elements should be scaled for a 24x24px viewport.
+If you need a custom SVG icon (not available in the Material Icons [default set](/components/material-icons/)) you can use the `SvgIcon` wrapper.
+The component extends the native `<svg>` element:
 
-The resulting icon can be used as is,
-or included as a child for other Material-UI components that use icons.
-By default, an Icon will inherit the current text color.
-Optionally, you can set the icon color using one of the theme color properties: `primary`, `secondary`, `action`, `error` & `disabled`.
+- It comes with built-in accessibility.
+- SVG elements should be scaled for a 24x24px viewport, so the resulting icon can be used as is, or included as a child for other Material-UI components that use icons (can be customized with the `viewBox` attribute).
+- By default, the component inherits the current color.
+Optionally, you can set the icon color using one of the theme color prop.
+
+```jsx
+function HomeIcon(props) {
+  return (
+    <SvgIcon {...props}>
+      <path d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z" />
+    </SvgIcon>
+  );
+}
+```
+
+### Color
+
+{{"demo": "pages/components/icons/SvgIconsColor.js"}}
+
+### Size
 
-{{"demo": "pages/components/icons/SvgIcons.js"}}
+{{"demo": "pages/components/icons/SvgIconsSize.js"}}
+
+### Component prop
```

Comment:
The controversial part.

Reply:
Controversial?

Reply:
There is an ongoing discussion in #17287 regarding removing the support for this prop.